prow/tide/tide.go

```diff
@@ -678,7 +701,8 @@ func (c *Controller) pickBatch(sp subpool) ([]PullRequest, error) {
 		if i == 5 {
 			break
 		}
-		if !isPassingTests(sp.log, c.ghc, pr) {
+		bp := c.ca.Config().Tide.BranchProtection.GetBranchProtection(sp.org, sp.repo, sp.branch)
+		if !isPassingTests(sp.log, c.ghc, pr, bp) {
```

Comment:
`pickBatch` should accept `bp` as a parameter from its caller instead of looking it up from the config.